File: force-field/python/particle.py
from vector import Vector

class Particle:
  def __init__(self, pos, mass):
    self.position = pos
    self.mass = mass
    self.velocity = Vector(0, 0)
  
  def forceFrom(self, p):
    distance = p.position.subtract(self.position)
    magnitude = self.mass * p.mass / distance.dot(distance)
    return distance.normalize().scale(magnitude)
  
  def applyForce(self, force, timeDelta):
    self.velocity.addInPlace(force.scale(timeDelta / abs(self.mass)))
    self.position.addInPlace(self.velocity.scale(timeDelta))
  
  def __str__(self):
    return ("{Particle mass: " + str(self.mass) + " velocity: " +
      str(self.velocity) + " position: " + str(self.position) + "}")

File: force-field/cs/Particle.cs
using System;

namespace ForceField {

  public class Particle {
    
    public Vector position;
    public double mass;
    public Vector velocity;
    
    public Particle(Vector pos, double _mass) {
      position = pos;
      mass = _mass;
      velocity = new Vector();
    }
    
    public Vector forceFrom(Particle p) {
      Vector distance = p.position.subtract(position);
      double magnitude = mass * p.mass / (distance.dot(distance));
      return distance.normalize().scale(magnitude);
    }
    
    public void applyForce(Vector force, double timeDelta) {
      velocity.addInPlace(force.scale(timeDelta / Math.Abs(mass)));
      position.addInPlace(velocity.scale(timeDelta));
    }
    
    public override string ToString() {
      return "{Particle mass: " + mass + " velocity: " + velocity +
        " position: " + position + "}";
    }
    
  }
  
}
